bokeh/client/connection.py

```diff
@@ -240,18 +266,21 @@ async def _connect_async(self):
         try:
             socket = await websocket_connect(request, subprotocols=["bokeh", self._session.token])
             self._socket = WebSocketClientConnectionWrapper(socket)
+        except HTTPClientError as e:
+            await self._transition_to_disconnected(DISCONNECTED(ErrorReason.HTTP_ERROR, e.code, e.message))
+            return
         except Exception as e:
             log.info("Failed to connect to server: %r", e)
 
         if self._socket is None:
-            await self._transition_to_disconnected()
+            await self._transition_to_disconnected(DISCONNECTED(ErrorReason.NETWORK_ERROR, None, "Socket invalid."))
```

Comment:
The comment above states 
> this property holds the error id. 0 otherwise. 
but None is passed here. I am fine with either tho I think I prefer setting None when there is no error code. But either here or the docstring should be updated to match.